aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/ObjectTaggingProvider.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.services.s3.transfer;
+
+import com.amazonaws.services.s3.model.ObjectTagging;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * This is the callback interface which is used by TransferManager.uploadDirectory {@link TransferManager#uploadDirectory(String, String, File, boolean, ObjectMetadataProvider, ObjectTaggingProvider)} and
```

Comment:
Some minor things:
- Remove "TransferManager.uploadDirectory" and "TransferManager.uploadFileList" and just leave the `{@link}`s
- The `@link` for uploadDirectory is the wrong overload, it should be the new one with the ObjectTaggingProvider param
- wrap the javadoc 100 characters per line